### DISCUSSION TITLE
[Feature Request] Dynamic Theme changing based on os-parameters

### DISCUSSION THREAD

**Discussion by zDEFz** [stoatchat/discussions#487] [Feature Suggestions]
### What do you want to see?

# Dynamic Theme changing based on os-parameters
- https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme
- For Accessibility, find a way to also detect high-contrast-white and high-contrast-black themes.

### PWA

- [ ] Yes, this feature request is specific to the PWA.

**Comment by Rexogamer** [comment]
some thoughts:
- this should be a third theme option (i.e. dark, light, match)
- enabled by default for new users? use detected scheme as default??

**Comment by Freeplayg** [comment]
For the contrast bit, there's https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast
However, it still isn't supported in Firefox yet (they have a flag you can enable in about:config).

I do think the match/auto option is a good idea, as I sometimes switch between light and dark OS themes.

**Comment by liquidnya** [comment]
I would love that for the website and also the "Find new servers to join" page and also for source code highlighting.
I have a lot of issues reading white text on dark backgrounds, and so navigating the website and looking at source code snippets that friends send me is really difficult for me.
So I have light mode enabled everywhere (even though I used to prefer dark mode), but there are still a lot of apps and sites forcing dark mode.
I am not sure if I should create a separate suggestion or if this feedback belongs to this thread.
